src/WebApi/Infrastructure/Middleware/BasicAuthMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using System.Net.Http.Headers;
+
+namespace Sindibad.SAD.WebTemplate.WebApi.Infrastructure.Middleware
+{
+    public class BasicAuthMiddleware
```

Comment:
I know middlewares don't need to implement any interfaces, but I think its clarity-wise to implement the `IMiddleware` interface